projects/igniteui-angular/src/lib/icon/icon.service.ts

```diff
@@ -102,7 +134,90 @@ export class IgxIconService {
      * ```
      */
     public familyClassName(alias: string): string {
-        return this._familyAliases.get(alias) || alias;
+        return this._families.get(alias)?.className || alias;
+    }
+
+    /** @hidden @internal */
+    private familyType(alias: string): IconType {
+        return this._families.get(alias)?.type;
+    }
+
+    /**
+     *  Creates a family to className relationship that is applied to the IgxIconComponent
+     *   whenever that family name is used.
+     * ```typescript
+     *   this.iconService.setFamily('material', { className: 'material-icons', type: 'liga' });
+     * ```
+     */
+    public setFamily(name: string, meta: FamilyMeta) {
+        this._families.set(name, meta);
+    }
+
+    /**
+     *  Adds an icon reference meta for an icon in a meta family.
+     *  Executes only if no icon reference is found.
+     * ```typescript
+     *   this.iconService.addIconRef('aruba', 'default', { name: 'aruba', family: 'svg-flags' });
+     * ```
+     */
+    public addIconRef(name: string, family: string, icon: IconMeta) {
+        const iconRef = this._iconRefs.get(family)?.get(name);
+
+        if (!iconRef) {
+            this.setIconRef(name, family, icon, false);
+        }
+    }
+
+    /**
+     *  Similar to addIconRef, but always sets the icon reference meta for an icon in a meta family.
+     * ```typescript
+     *   this.iconService.setIconRef('aruba', 'default', { name: 'aruba', family: 'svg-flags' });
+     * ```
+     */
+    public setIconRef(name: string, family: string, icon: IconMeta, update = true) {
```

Comment:
```suggestion
    public setIconRef(name: string, family: string, icon: IconMeta) {
```
Let's leave this without the extra parameter.